From f307864f904062a107c829db075ac022dd7cbd5b Mon Sep 17 00:00:00 2001
From: Tim Lievens <timlievens@gmail.com>
Date: Tue, 15 Jan 2019 09:48:52 +0100
Subject: [PATCH 1/1] Changed maxlength

---
 custom_links.admin.inc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/custom_links.admin.inc b/custom_links.admin.inc
index 8cb53db..80f1467 100644
--- a/custom_links.admin.inc
+++ b/custom_links.admin.inc
@@ -76,6 +76,7 @@ function custom_links_form() {
   $form['link']['querystring'] = array(
     '#type' => 'textfield',
     '#title' => t('Querystring'),
+    '#maxlength' => 256,
     '#description' => t("The optional querystring for the link. (!sample_url)", array('!sample_url' => $base_url.'/article?<u><b>id=1</b></u>')),
     '#default_value' => $lid ? $link->query : NULL
   );
-- 
2.14.3 (Apple Git-98)

